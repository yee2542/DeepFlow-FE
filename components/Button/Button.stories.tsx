// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import CustomButton from 'Components/Button/CustomButton';
import { ButtonComponentProps } from 'Components/Button/button.type';
import React from 'react';

export default {
  title: 'DeepFlow/Button',
  component: CustomButton,
} as Meta;

const Template: Story<ButtonComponentProps> = (args) => <CustomButton {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  style: 'primary',
  children: 'Button',
} as ButtonComponentProps;

export const Secondary = Template.bind({});
Secondary.args = {
  style: 'secondary',
  children: 'Button',
} as ButtonComponentProps;

export const Danger = Template.bind({});
Danger.args = {
  style: 'danger',
  children: 'Button',
} as ButtonComponentProps;

export const Success = Template.bind({});
Success.args = {
  style: 'success',
  children: 'Button',
} as ButtonComponentProps;

export const Disable = Template.bind({});
Disable.args = {
  style: 'disable',
  children: 'Button',
};

export const Small = Template.bind({});
Small.args = {
  style: 'primary',
  size: 'small',
  children: 'Button',
};

export const Medium = Template.bind({});
Medium.args = {
  style: 'primary',
  size: 'medium',
  children: 'Button',
};

export const Large = Template.bind({});
Large.args = {
  style: 'primary',
  size: 'large',
  children: 'Button',
};
