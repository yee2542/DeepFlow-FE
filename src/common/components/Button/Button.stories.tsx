import { Meta, Story } from '@storybook/react';
import Button from 'Components/Button';
import { ButtonComponentProps } from 'Components/Button/types';
import React from 'react';
import { GlobalStyle } from 'Styles/global';

export default {
  title: 'FluentSearch/Button',
  component: Button,
} as Meta;

const ButtonStory: Story<ButtonComponentProps> = (args) => (
  <>
    <GlobalStyle />
    <Button {...args} />
  </>
);

// Color

export const Default = ButtonStory.bind({});
Default.args = {
  children: 'Button',
};

export const Secondary = ButtonStory.bind({});
Secondary.args = {
  color: 'secondary',
  children: 'Button',
};

export const Danger = ButtonStory.bind({});
Danger.args = {
  color: 'danger',
  children: 'Button',
};

export const Success = ButtonStory.bind({});
Success.args = {
  color: 'success',
  children: 'Button',
};

export const Disable = ButtonStory.bind({});
Disable.args = {
  color: 'disable',
  children: 'Button',
};

// Size

export const Small = ButtonStory.bind({});
Small.args = {
  color: 'primary',
  size: 'small',
  children: 'Button',
};

export const Large = ButtonStory.bind({});
Large.args = {
  color: 'primary',
  size: 'large',
  children: 'Button',
};
