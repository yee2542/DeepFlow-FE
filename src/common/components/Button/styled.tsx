import { ButtonComponentProps } from 'Components/Button/types';
import styled, { css } from 'styled-components';

const colorMapping = (color?: ButtonComponentProps['color']) => {
  switch (color) {
    case 'secondary':
      return css`
        background-color: #fff;
        color: var(--green-color);
        border: 1px solid var(--green-color);
        &:hover {
          color: var(--light-green-color);
          border: 1px solid var(--light-green-color);
        }
      `;
    case 'danger':
      return css`
        background-color: var(--error-color);
        color: #fff;
        &:hover {
          background-color: var(--light-error-color);
        }
      `;
    case 'success':
      return css`
        background-color: var(--success-color);
        color: #fff;
        &:hover {
          background-color: var(--light-success-color);
        }
      `;
    case 'disable':
      return css`
        background-color: var(--gray-color);
        color: var(--dark-gray-color);
        border: 1px solid var(--dark-gray-color);
        &:hover {
          cursor: not-allowed;
        }
      `;
    // default -> primary
    default:
      return css`
        background-color: var(--green-color);
        color: #fff;
        &:hover {
          background-color: var(--light-green-color);
        }
      `;
  }
};

const sizeMapping = (size?: ButtonComponentProps['size']) => {
  switch (size) {
    case 'small':
      return css`
        font-size: var(--small-body);
        padding: 10px 16px;
      `;
    case 'large':
      return css`
        font-size: var(--h5);
        padding: 12px 24px;
      `;
    // default -> medium
    default:
      return css`
        font-size: var(--body);
        padding: 11px 20px;
      `;
  }
};

export const ButtonStyle = styled.button<ButtonComponentProps>`
  border-radius: 100px;
  border: 0px;
  ${({ color }) => colorMapping(color)};
  ${({ size }) => sizeMapping(size)};
  cursor: pointer;
  font-weight: bold;
  &:focus {
    outline: 0;
    box-shadow: none !important;
  }
`;
