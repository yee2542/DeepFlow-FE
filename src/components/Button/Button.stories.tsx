// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import Button from 'Components/Button';
import { ButtonComponentProps } from 'Components/Button/type';
import React from 'react';

export default {
  title: 'FluentSearch/Button',
  component: Button,
} as Meta;

const ButtonStory: Story<ButtonComponentProps> = (args) => <Button {...args} />;

// Color

export const Default = ButtonStory.bind({});
Default.args = {
  children: 'Button',
};

export const Secondary = ButtonStory.bind({});
Secondary.args = {
  color: 'secondary',
  children: 'Button',
};

export const Danger = ButtonStory.bind({});
Danger.args = {
  color: 'danger',
  children: 'Button',
};

export const Success = ButtonStory.bind({});
Success.args = {
  color: 'success',
  children: 'Button',
};

export const Disable = ButtonStory.bind({});
Disable.args = {
  color: 'disable',
  children: 'Button',
};

// Size

export const Small = ButtonStory.bind({});
Small.args = {
  color: 'primary',
  size: 'small',
  children: 'Button',
};

export const Large = ButtonStory.bind({});
Large.args = {
  color: 'primary',
  size: 'large',
  children: 'Button',
};
