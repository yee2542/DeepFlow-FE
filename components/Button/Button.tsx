import { CustomButton } from 'Components/Button/Button.styled';
import { ButtonComponentProps } from 'Components/Button/button.type';
import React from 'react';

const Button: React.FC<ButtonComponentProps> = (props) => {
  const { children, style, color, size } = props;

  return (
    <CustomButton color={color} size={size} style={style}>
      {children}
    </CustomButton>
  );
};

export default Button;
