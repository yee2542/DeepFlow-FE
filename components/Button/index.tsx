import { CustomButton } from 'Components/Button/styled';
import { ButtonComponentProps } from 'Components/Button/type';
import React from 'react';

const Button: React.FC<ButtonComponentProps> = (props) => {
  const { children, style, color, size } = props;

  return (
    <CustomButton color={color} size={size} style={style}>
      {children}
    </CustomButton>
  );
};

export default Button;
