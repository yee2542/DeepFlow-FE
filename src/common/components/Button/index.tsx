import { ButtonStyle } from 'Components/Button/styled';
import { ButtonComponentProps } from 'Components/Button/types';
import React from 'react';

const Button: React.FC<ButtonComponentProps> = (props) => {
  const { children, style, color, size } = props;

  return (
    <ButtonStyle color={color} size={size} style={style}>
      {children}
    </ButtonStyle>
  );
};

export default Button;
