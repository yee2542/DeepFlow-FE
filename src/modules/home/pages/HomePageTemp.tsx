import React from 'react';
import { HomeTempWrapper, ContentTempWrapper } from './styled';
import Image from 'next/image';
import Button from 'Components/Button';
import Link from 'next/link';

const HomePageTemp: React.FC = () => {
  return (
    <>
      <HomeTempWrapper>
        <ContentTempWrapper>
          <Image src="/assets/images/logo-home.svg" width={800} height={400} />
          <br />
          <Link href="/register">
            <a>
              <Button
                style={{
                  backgroundColor: 'white',
                  color: '#5A36CC',
                  padding: '15px 30px 15px 30px',
                }}>
                Get Started
              </Button>
            </a>
          </Link>
        </ContentTempWrapper>
      </HomeTempWrapper>
    </>
  );
};

export default HomePageTemp;
