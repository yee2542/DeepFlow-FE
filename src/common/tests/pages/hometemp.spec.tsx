import { mount, shallow } from 'enzyme';
import React from 'react';
import HomePageTemp from 'Modules/home/pages/HomePageTemp';
import Button from 'Components/Button';

describe('Pages/HomeTemp test', () => {
  it('Render correctly', () => {
    const wrap = shallow(<HomePageTemp />);
    expect(wrap.html).toMatchSnapshot();
  });

  it('HomeTemp should be exisiting', () => {
    const wrap = shallow(<HomePageTemp />);
    expect(wrap.exists()).toBe(true);
  });

  it('Should contain Get Started Button', () => {
    const wrap = mount(<HomePageTemp />);
    expect(wrap.find(Button).text()).toEqual('Get Started');
  });

  it('Should contain Logo Image', () => {
    const wrap = mount(<HomePageTemp />);
    expect(wrap.exists('img')).toBe(true);
  });
});
