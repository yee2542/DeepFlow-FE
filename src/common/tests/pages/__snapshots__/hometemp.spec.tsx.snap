// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/HomeTemp test Render correctly 1`] = `[Function]`;
